src/UI/Implementation/Component/Button/Toggle.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/* Copyright (c) 2018 Thomas Famula <famula@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Button;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Component\Signal;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+use ILIAS\UI\Implementation\Component\JavaScriptBindable;
+use ILIAS\UI\Implementation\Component\Triggerer;
+
+class Toggle extends Button implements C\Button\Toggle {
+	use ComponentHelper;
+	use JavaScriptBindable;
+	use Triggerer;
+
+	/**
+	 * @var string|null
+	 */
+	protected $action_deactivated;
+
+	public function __construct($label, $action, $action_deactivated)
+	{
+		parent::__construct($label, $action);
+
+		$this->checkArg(
+			"action_deactivated",
+			is_string($action_deactivated) || $action_deactivated instanceof Signal,
+			$this->wrongTypeMessage("string or Signal", gettype($action_deactivated))
```

Comment:
We will use this a lot in the future, so i would very much appreciate if you put this into a `checkStringOrSignal`-helper method on the `ComponentHelper`.

Reply:
Done. I also changed the method in the `Button.php` class. This was the only other place I found, where this code was used.